application/include/mmul.hpp

```diff
@@ -38,22 +38,32 @@ void mmul(Executor ex, MatrixXd &a, MatrixXd &b, MatrixXd &c) {
       }
     }
   };
-  ex.bulk_execute(mul, 1, a, b, c);
+  ex.bulk_execute(mul, a.rows());
 }
 
 template <typename Executor,
-    typename execution::executor_of_type<cuda_executor, Executor> = 0>
+          typename execution::instance_of_base<cuda_executor, Executor> = 0>
 void mmul(Executor ex, MatrixXd &a, MatrixXd &b, MatrixXd &c) {
+  double *a_g, *b_g, *c_g;
+  a_g = static_cast<double *>(
+      device_upload(static_cast<void *>(a.data()), a.size() * sizeof(double)));
+  b_g = static_cast<double *>(
+      device_upload(static_cast<void *>(b.data()), b.size() * sizeof(double)));
+  c_g = static_cast<double *>(
+      device_upload(static_cast<void *>(c.data()), c.size() * sizeof(double)));
```

Comment:
I think you only need the static cast for the return type, and you can initialize your pointer directly.

Reply:
Fixed in #12